### PR TITLE
Update Cardano.Api.Tx in preparation for Alonzo

### PR DESCRIPTION
**Pull request by Jimbo4350**
Depends on: 
1. #2657
2. #2400 
3. #2699
4. #2649

### CODE REVIEW

File: cardano-api/src/Cardano/Api/TxBody.hs [hunk 1]
```diff
@@ -878,6 +878,9 @@ data TxBody era where
           -- witnesses set, since they need to be known when building the body.
        -> [Ledger.Script (ShelleyLedgerEra era)]
 
+          --TODO: we will probably want to or need to put the Alonzo data and
```
Comment (Jimbo4350) [inline cardano-api/src/Cardano/Api/TxBody.hs:899]: :+1:

**Review by newhoggy — APPROVED** [review]

**Comment by newhoggy** [comment]
bors merge

**Review by dcoutts — APPROVED** [review]
I'm happy with this now.

Follow-up commits are now in #2738, in particular the redeemer map stuff.

**Comment by dcoutts** [comment]
bors merge

**Pull request merged**